sdk/tables/Azure.Data.Tables/src/Extensions/DictionaryTableExtensions.cs

```diff
@@ -12,6 +13,11 @@ namespace Azure.Data.Tables
 {
     internal static class DictionaryTableExtensions
     {
+        /// <summary>
+        /// A cache for reflected <see cref="PropertyInfo"/> array for the given <see cref="Type"/>.
+        /// </summary>
+        private static readonly ConcurrentDictionary<Type, PropertyInfo[]> s_propertyInfoCache = new ConcurrentDictionary<Type, PropertyInfo[]>();
```

Comment:
Should we limit the size of the cache?

Reply:
Good question - I suppose we shouldn't make it unbounded.

Reply:
Thinking more about this, I'm wondering if it would be worth while to implement a cache eviction strategy. I think a significantly large set of different strongly typed entities might be an edge case.